File: 3974MaximumTotalSumofKSelectedElements.py
class Solution(object):
    def maxSum(self, nums, k, mul):
        """
        :type nums: List[int]
        :type k: int
        :type mul: int
        :rtype: int
        """
        nums.sort(reverse=True)
        n = len(nums)
        val = min(n,k)
        res = 0
        for i in range(val):
            if mul > 0:
                res += nums[i]*mul
                mul -= 1
            else:
                res += nums[i]

        return res
            
        
        

File: 3974_MaximumTotalSumofkSelectedElements.cpp
class Solution {
public:
    long long maxSum(vector<int>& nums, int k, int mul) {
        sort(nums.begin(), nums.end(), greater<int>());

        long long ans = 0;

        for (int i = 0; i < k; i++) {
            long long factor = max(1, mul - i);
            ans += 1LL * nums[i] * factor;
        }

        return ans;
    }
};